ios_sample_test_app/ios_sample_test_app/ViewController.swift

```diff
@@ -0,0 +1,488 @@
+        button.setTitle("Change Background Color", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(changeBackgroundColor), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private lazy var buyNowButton: UIButton = {
+       let button = UIButton()
+        button.backgroundColor = UIColor(rgb: 0x0074DF)
+        button.setTitle("Buy Now", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(sendPurchaseEvent(_:)), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private lazy var addToCartButton: UIButton = {
+       let button = UIButton()
+        button.backgroundColor = UIColor(rgb: 0x0074DF)
+        button.setTitle("Add To Cart", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(sendAddToCartEvent(_:)), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private lazy var colorBlockPageButton: UIButton = {
+        let button = UIButton()
+        button.backgroundColor = UIColor(rgb: 0x0074DF)
+        button.setTitle("Color Block Page", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(openColorBlockPage(_:)), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private lazy var readDeepLinkButton: UIButton = {
+       let button = UIButton()
+        button.backgroundColor = UIColor(rgb: 0x0074DF)
+        button.setTitle("Read Link Data", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(openReadDeepLinkPage(_:)), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
```

Comment:
Since these buttons share a lot of the same properties, you could create a simple function returns a UIButton and takes a title string and action selector.

Reply:
Created function that takes in the button title and sets both title and action based on the title